Mathlib/Algebra/MvPolynomial/SchwartzZippel.lean

```diff
@@ -0,0 +1,201 @@
+      #{x ∈ S ^^ (n + 1) | eval x p = 0} / ∏ i, (#(S i) : ℚ≥0)
+          -- In the first case, `pₖ` evaluates to `0`.
+        = #{x ∈ S ^^ (n + 1) | eval x p = 0 ∧ eval (tail x) pₖ = 0} / ∏ i, (#(S i) : ℚ≥0)
+          -- In the second case, `pₖ` does not evaluate to `0`.
+          + #{x ∈ S ^^ (n + 1) | eval x p = 0 ∧ eval (tail x) pₖ ≠ 0} / ∏ i, (#(S i) : ℚ≥0) := by
+        rw [← add_div, ← Nat.cast_add, ← card_union_add_card_inter, filter_union_right,
+          ← filter_and]
+        simp [← and_or_left, em, and_and_and_comm]
+      _ ≤ (pₖ.support.sup fun s ↦ ∑ i, (s i / #(S i.succ) : ℚ≥0)) + p.degreeOf 0 / #(S 0) := ?_
+      -- _ = (pₖ.support.sup fun s ↦ ∑ i, (s i / #(S i.succ) : ℚ≥0)) + k / #(S 0) := ?_
+      _ ≤ p.support.sup fun s ↦ ∑ i, (s i / #(S i) : ℚ≥0) := ?_
+    · gcongr ?_ + ?_
+      · -- We bound the size of the first set by induction
+        calc
+          #{x ∈ S ^^ (n + 1) | eval x p = 0 ∧ eval (tail x) pₖ = 0} / ∏ i, (#(S i) : ℚ≥0)
+            ≤ #{x ∈ S ^^ (n + 1) | eval (tail x) pₖ = 0} / ∏ i, (#(S i) : ℚ≥0) := by
+            gcongr; exact fun x hx ↦ hx.2
+          _ = #(S 0) * #{xₜ ∈ tail S ^^ n | eval xₜ pₖ = 0}
+              / (#(S 0) * (∏ i, #(S (.succ i)) : ℚ≥0)) := by
+            rw [card_consEquiv_filter_piFinset S fun x ↦ eval x pₖ = 0, prod_univ_succ, tail_def]
+            norm_cast
+          _ ≤ #{xₜ ∈ tail S ^^ n | eval xₜ pₖ = 0} / ∏ i, (#(S (.succ i)) : ℚ≥0) :=
+            mul_div_mul_left_le (by positivity)
+          _ ≤ (pₖ.support.sup fun s ↦ ∑ i, (s i / #(S (.succ i)) : ℚ≥0)) :=
+            schwartz_zippel_sup_sum hpₖ₀ _
+      · -- We bound the second set by noting that if `x` is in it, then `x₀` is the root of
+        -- the univariate polynomial`pₓ` obtained by evaluating each (multivariate polynomial)
+        -- coefficient at `xₜ`. Since `pₓ` has degree `k`, there are at most `k` such `x₀` for
+        -- each `xₜ`, which gives the result.
+        calc
+          #{x ∈ S ^^ (n + 1) | eval x p = 0 ∧ eval (tail x) pₖ ≠ 0} / ∏ i, (#(S i) : ℚ≥0)
+            ≤ ↑(p.degreeOf 0 * ∏ i, #(S (.succ i))) / ∏ i, (#(S i) : ℚ≥0) := ?_
+          _ = p.degreeOf 0 * (∏ i, #(S (.succ i))) / (#(S 0) * ∏ i, #(S (.succ i))) := by
+            norm_cast; rw [prod_univ_succ]
+          _ ≤ (p.degreeOf 0 / #(S 0) : ℚ≥0) := mul_div_mul_right_le (by positivity)
+        gcongr
+        calc
+          #{x ∈ S ^^ (n + 1) | eval x p = 0 ∧ eval (tail x) pₖ ≠ 0}
+            = #{x ∈ S ^^ (n + 1) | eval (tail x) pₖ ≠ 0 ∧ eval x p = 0} := by simp_rw [and_comm]
+          _ = #({xₜ ∈ tail S ^^ n | eval xₜ pₖ ≠ 0}.biUnion fun xₜ ↦ image (fun x₀ ↦ (x₀, xₜ))
+                {x₀ ∈ S 0 | eval (cons x₀ xₜ) p = 0}) := by
+            rw [← filter_filter, filter_piFinset_eq_map_consEquiv S (fun r ↦ eval r pₖ ≠ 0),
+              filter_map, card_map, product_eq_biUnion_right, filter_biUnion]
+            simp [Function.comp_def, filter_image, filter_filter]
+            rfl
```

Comment:
Why do you need this rfl? Missing lemma?

Reply:
Yes, we're missing `coe_consEquiv`, which should come from `simps` but `simps` is currently misconfigured and I am planning to fix it in another PR